pkg/cli/deployment/deploy.go

```diff
@@ -49,7 +49,7 @@ const (
 
 type ResourceDeploymentClient struct {
 	RadiusResourceGroup string
-	Client              *sdkclients.ResourceDeploymentsClient
+	Client              sdkclients.ResourceDeploymentsClient
```

Comment:
I changed this from a struct to an interface (since I use the same type in deploymenttemplate controller). This makes testing way easier.